all-of-us/workbench
# [RW-707][risk=low] Update Account Success Page

Addresses https://precisionmedicineinitiative.atlassian.net/browse/RW-707

## Changes
* Added resend email feature
* Added change contact email feature

## Notes
Both features do not use authentication because the user has not authenticated yet and needs access to both. Resend is benign and has no impact. We have a very narrow security risk with the change contact email feature. In this scenario, we are only allowing the change of contact email when the user has never signed in. This feature will not work for users that have previously signed in.

## Testing
* Create a new account in the usual manner
* Check your contact email for the welcome email
* Before using the account, stay on the success page and ...
  * Try resending the welcome email and look for it in your email
  * Try changing contact email and look for the welcome message under the new account.
* If you want to be tricky, craft a curl call to change your contact email for a user that has signed in. This fails with a 403 error.
* Before actually using this account, you can delete it in gsuite and re-use the same information to re-test.

## Welcome Message
Don't worry - that password is not valid.
![screen shot 2018-07-03 at 10 54 27 am](https://user-images.githubusercontent.com/116679/42227442-894515a4-7eaf-11e8-9c95-04339c7c7fbe.png)

## Resend Modal
![screen shot 2018-07-03 at 9 39 50 am](https://user-images.githubusercontent.com/116679/42229558-c086f078-7eb4-11e8-9947-5189f0c88390.png)

## Change Contact Email Modal
![screen shot 2018-07-03 at 9 40 05 am](https://user-images.githubusercontent.com/116679/42229584-cbf84650-7eb4-11e8-8967-dabbf2827d1f.png)
